core/app/models/spree/return_authorization.rb

```diff
@@ -61,6 +62,10 @@ def can_cancel_return_items?
 
     private
 
+      def authorize_return
+        order.authorize_return! unless order.nil?
```

Comment:
it seems to me that this is a stretched domain design. Upon the creation of a return_authorization record, you are invoke a callback to (possibly) flip the state of the order to 'awaiting_return'. While it makes sense on the surface, a return_authorization object must now have domain knowledge about an order object. 

Does it need to have this knowledge?  Seems like a tight dependency that doesn't need to be there.  (As well, I think we should move away from the over-use of callbacks generally throughout Spree, and this just adds another one)

A cleaner domain design would be to allow the creation of a return_authorization only through a a clean API on the Order, like a  ```ReturnAuthorizationCreator``` object that receives an Order, some information about the RMA to create, etc. 

It is then this new object's responsibility to both flip the state of the order AND, if allowed, create the return_authorization records, failing if necessary conditions aren't met.

This way, you encapsulate the rules for _what_ and _how_ messages are sent between objects and remove knowledge about the Order object away from the ReturnAuthorization object